src/Drupal/DrupalExtension/Context/DrupalContext.php

```diff
@@ -183,32 +183,135 @@ public function getTableRow(Element $element, $search)
         throw new \Exception(sprintf('Failed to find a row containing "%s" on the page %s', $search, $this->getSession()->getCurrentUrl()));
     }
 
+  /**
+   * Retrieve table rows containing specified text from a given element.
+   *
+   * @param Element $element
+   *   \Behat\Mink\Element\Element object.
+   * @param string $search
+   *   The text to search for in the table row.
+   *
+   * @return array of \Behat\Mink\Element\NodeElement
+   *
+   * @throws \RuntimeException
+   */
+  public function getTableRows(Element $element, $search): array {
+    $rows = $element->findAll('css', 'tr');
+    if (empty($rows)) {
+      throw new \RuntimeException(sprintf('No rows found on the page %s', $this->getSession()->getCurrentUrl()));
+    }
+    $found_rows = [];
+    foreach ($rows ?: [] as $row) {
+      if (strpos($row->getText(), $search) !== FALSE) {
+        array_push($found_rows, $row);
+      }
+    }
+    if (count($found_rows) === 0) {
+      throw new \RuntimeException(sprintf('Failed to find a row containing "%s" on the page %s', $search, $this->getSession()->getCurrentUrl()));
+    }
+    return $found_rows;
+  }
+
+  /**
+   * See if the element has a table.
+   *
+   * @param Element $element
+   *    \Behat\Mink\Element\Element object.
+   *
+   * @return bool
+   *   True if the element has a table.
+   */
+  private function hasTable(Element $element): bool {
+    $rows = $element->findAll('css', 'tr');
+    return !empty($rows);
+  }
+
   /**
    * Find text in a table row containing given text.
    *
    * @Then I should see (the text ):text in the :rowText row
    */
-    public function assertTextInTableRow($text, $rowText)
-    {
-        $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
-        if (strpos($row->getText(), $text) === false) {
-            throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+    public function assertTextInTableRow($text, $rowText) {
```

Comment:
@kumikoono Let's add a return type hint:

```suggestion
    public function assertTextInTableRow($text, $rowText): void {
```